scalingbitcoin/tel-aviv-2019/edgedevplusplus/2019-09-09-carla-kirk-cohen-routing-problems-and-solutions.md

```diff
@@ -19,39 +19,49 @@ Video: <https://youtu.be/1O-bhcbh9vE>
+These messages also indicate nodes' willingness to accept inbound connections, so IPv4 or 6, or Tor addresses, which aren't really related to routing but it does provide you with some additional information.
+
+And finally these announcements have a strictly increasing timestamp. So if we see a timestamp with an old node announcement, a node announcement that we've already seen a newer timestamp for, and we've seen all the timestamps, we assume something is wrong and we disconnect from this node because they shouldn't be behaving like this.
 
 ## Channel updates
 
-And finally we look at channel updates so with no announcements and channel announcements we can build what the graph looks like. Channel updates allow us to change the values of this graph so specifically you can update your channel policy. This is the fees that you wish to receive for routing over this edge which is a base fee per every single payment and then a fee paid per Satoshi thereafter and that's a bit of a difference to Bitcoin. 
+And finally we look at channel updates. With node announcements and channel announcements we can build what the graph looks like. Channel updates allow us to change the values this graph has. So specifically, you can update your channel policy. This is the fees that you wish to receive for routing over this edge, which is a base fee per every single payment and then a fee paid per Satoshi thereafter.
+
+And that's a bit of a difference to Bitcoin that people don't often realize when they start to look at Lightning, is that your fee does scale with the size of your transaction (meaning the Bitcoin size, not the kilobyte size), so that's a big difference.
+
+This (the channel update) also contains the time-lock information that you wish to accept over this channel and the limits that you'll pay some HTLCs: the minimum amounts, the maximum amounts, and a total number of HTLCs you're willing to route through this channel. And these again are set per node in a channel, so a channel has direction--you can move funds back and forth between the two peers--and each node has the right to determine their own fees in their own time lock according to their risk preference.
+
+These messages can also be used to keep your node alive. If you haven't done anything for two weeks, your tunnel will be pruned from the graph in most implementations, so you can send out one of these to keep alive. However, these messages are currently flooded and they make up a very large portion of network traffic, almost 40% last time I looked, which was about two months ago. This is generally because nodes are flipping online and offline a lot. So every time they come back online they send out a new channel update message to indicate what their current rates are, and this really clogs up the network a lot. It's a bit of an issue at the moment.
```

Comment:
Note to self: realized after that "flapping" rather than "flipping" is the correct word, so should revert that word in the follow-up.